tests/app/test_model.py

```diff
@@ -894,6 +896,120 @@ def test_lot_data_is_serialized(self):
+
+    def test_no_suppliers(self, open_example_framework):
+        """Test a framework with no suppliers on it returns no submitted services."""
+        with self.app.app_context():
+            f = Framework.query.filter(Framework.id == open_example_framework['id']).first()
+            supplier_ids_with_completed_service = f.get_supplier_ids_for_completed_service()
+            assert len(supplier_ids_with_completed_service) == 0
+
+
+class TestFrameworkSupplierIdsMany(BaseApplicationTest):
+    """Test multiple suppliers, multiple services."""
+
+    def setup(self):
+        super(TestFrameworkSupplierIdsMany, self).setup()
+        # We need to clear these up at the end, set an instance attribute here.
+        self.service_ids = []
+
+    def test_multiple_services(self, open_example_framework):
+        with self.app.app_context():
+            lot = Lot.query.first()
+            self.fl_query = {
+                'framework_id': open_example_framework['id'],
+                'lot_id': lot.id
+            }
+            fl = FrameworkLot(**self.fl_query)
+            db.session.add(fl)
+            db.session.commit()
+
+            # Set u 5 dummy suppliers
+            self.setup_dummy_suppliers(5)
+            supplier_ids = range(5)
+            # 5 sets of statuses for their respective services.
+            service_status_choices = [
+                ('failed',),
+                ('not-submitted',),
+                ('failed', 'failed', 'failed', 'not-submitted', 'submitted', 'submitted', 'not-submitted', 'failed'),
+                (),
+                ('not-submitted', 'submitted', 'not-submitted', 'not-submitted', 'failed')
+            ]
+            # For the supplier, service_status sets create the services and draft services.
+            count = 0
+            for supplier_id, service_statuses in zip(supplier_ids, service_status_choices):
+                for service_status in service_statuses:
+                    service_id = str(count)*10
+                    self.service_ids.append(service_id)
+                    count += 1
+                    service = self.setup_dummy_service(service_id, supplier_id=supplier_id, **self.fl_query)
+
+                    db.session.commit()
+                    with mock.patch('app.models.url_for', autospec=lambda i, **values: 'test.url/test'):
+                        ds = DraftService.from_service(Service.query.filter(Service.service_id == service_id).first())
+                        ds.status = service_status
+                    db.session.add(ds)
+                    db.session.commit()
+            # Assert that only those suppliers whose service_status list contains failed and/ or submitted are returned.
+            framework = Framework.query.filter(Framework.id == open_example_framework['id']).first()
+            assert sorted(framework.get_supplier_ids_for_completed_service()) == [0, 2, 4]
+
+    def teardown(self):
```

Comment:
So the problem is that the db doesn't get wiped completely. As far as I understand it, test objects should be created with ids > 99. They'll then get wiped. It would be better to wipe everything though. Otherwise if you select a framework by slug for instance 
`dos = Framework.query.filter(Framework.slug == 'digital-outcomes-and-specialists').first()` we don't know what state it's going to be in or what's been done to it.